client/components/marketing-survey/cancel-purchase-form/index.jsx

```diff
@@ -678,9 +686,26 @@ class CancelPurchaseForm extends Component {
 			return translate( 'Remove product' );
 		}
 
-		if ( isPlan( purchase ) ) {
-			return translate( 'Cancel plan' );
+		if ( hasAmountAvailableToRefund( purchase ) ) {
+			if ( isDomainRegistration( purchase ) ) {
+				return translate( 'Cancel domain and refund' );
+			}
+			if ( isSubscription( purchase ) ) {
+				return translate( 'Cancel subscription' );
```

Comment:
If a refund is being processed, doesn't that mean the product is being removed?

If so, I think the language we want here would be more like "Cancel plan and refund" for a plan, and "Cancel product and refund" (by way of the wording further down) for anything else below, regardless of whether it's a subscription or one-time purchase?